tests/test_NSDictionary.py

```diff
@@ -51,10 +51,10 @@ def test_getitem(self):
     def test_iter(self):
         d = self.make_dictionary(self.py_dict)
 
-        keys = set(self.py_dict)
+        keys = {str(k) for k in self.py_dict}
         for k in d:
-            self.assertTrue(k in keys)
-            keys.remove(k)
+            self.assertTrue(str(k) in keys)
+            keys.remove(str(k))
```

Comment:
I get why this change is needed from an API point of view, but it feels like it's moved from being a check of NSString behavior, to a test of Python str behavior...

Reply:
I don't think it was ever meant as a test of `NSString` behavior... if it is, then it should be moved into `test_NSString`.

As I understand it, this test checks that `ObjCDictInstance`s can be iterated over correctly. To do that, it copies the Python dict's keys into a (mutable) set, then iterates over the ObjC dict and checks that every key appears exactly once in the iteration. If an incorrect or duplicate key appeared in the iteration, the `in` check would fail; if a key didn't appear in the iteration, the final `len` check would fail.

Reply:
The new iteration of this seems a more direct test of that behavior.